### DISCUSSION TITLE
Is it possible to use WebKitGTK with GTK4 built with gvsbuild?

### DISCUSSION THREAD

**Discussion by wooly17** [wingtk/gvsbuild#1265] [General]
Hello!  I've been using gvsbuild to build GTK4 for Windows/Python for years, and it works great - thanks for all the work you do!  

I was recently researching how to embed an interactive Google Maps window in a GTK4 Python app.  I don't know if this is possible, but while reading I came across some pretty old examples that either `import webkit` or do something like this (GTK3):

```
# Old example code found from 2020

import gi
gi.require_version('Gtk', '3.0')
gi.require_version('WebKit2', '4.0')

from gi.repository import Gtk, WebKit2
```

Is anything like this possible these days in GTK4 built for Python from gvsbuild?  For me, `'Gtk'` is the only namespace available in the `_versions` dictionary in gi's init file, and I didn't see anything related to webkit that I could import from gi.repository.  

I'd seen a blog post from about 11 months ago saying WebKitGTK is stable for GTK4, but I don't have the slightest clue how to build it for Windows or integrate it into my Python environment.

**Comment by danyeaw** [comment]
Hi @wooly17, this should be possible. You would have to follow the instructions here on how to build WebkitGTK: https://trac.webkit.org/wiki/BuildingGtk
Then create a new Gvsbuild CMake project using one of the others as a template and include any dependencies like gtk4.
